packages/widgets-manager/src/manager.ts

```diff
@@ -1,51 +1,130 @@
-import 'isomorphic-fetch';
-import { App, ObjectIndex } from "./types";
+import { makePostionStrategy } from './strategies/position/creator';
+import { diffBy } from './utils/index';
+import { POSITION_RELATIVE_TO_ELEMENT, POSITION_RELATIVE_TO_PLACE, POSITION_FIXED_TO_TOP } from './constants';
+import { GridManager } from './grid';
+import { App, ObjectIndex, GridCell } from "./types";
+import { ReplaceAppStrategy } from './strategies/mounting/replace';
 
-export class AppManager {
-  baseUrl: string;
-  urlParams: ObjectIndex;
+const diffByAppId = diffBy((x: App, y: App ) => x.id === y.id);
 
-  constructor(baseURL: string, params: ObjectIndex) {
-    this.baseUrl = baseURL;
-    this.urlParams = params;
+export class AppManager {
+  gridManager: GridManager;
+  fetchAppData: (appId: number) => Promise<any>;
+  
+  constructor(fetchAppData: (appId: number) => Promise<any>, gridManager: GridManager) {
+    this.fetchAppData = fetchAppData;
+    this.gridManager = gridManager;
   }
 
-  addStyleString = (cssRules: string) => {
+  _addStyleString = (app: App) => {
+    const { css } = app.settings;
     const node = document.createElement('style');
-    node.innerHTML = cssRules;
+    node.id = `letstalk-app-${app.id}-styles`;
+    node.innerHTML = css;
     document.body.appendChild(node);
   }
 
-  mountApp = (appId: Number, appOnload: () => void) => {
-    const iframe = document.createElement('iframe');
-    iframe.id = `letstalk-app-${appId}`;
-    iframe.onload = appOnload;
-    
-    const urlParams = Object.keys(this.urlParams).map(key => `${key}=${encodeURIComponent(this.urlParams[key])}`).join('&');
-    const widgetAppUrl = `${this.baseUrl}/api/v1/widget_apps/${appId}?${urlParams}`;
-    fetch(widgetAppUrl).then((widgetAppResonse) => {
-      widgetAppResonse.json().then((appConfiguration: App) => {
-        if (appConfiguration.payload_type === 'html') {
-          iframe.src = appConfiguration.payload;
-        }
-        // Add css style tag with style rules
-        this.addStyleString(appConfiguration.settings.css);
-        
-        // Apply settings to the iframe style property
-        Object.keys(appConfiguration.settings.inlineCss).forEach((key: string) => {
-          iframe.style.setProperty(key, appConfiguration.settings.inlineCss[key]);
+  _createIframeForApp = (app: App, cell: GridCell) => {
+    if (!document.getElementById(`letstalk-app-${app.id}`)) {
+      const iframe = document.createElement('iframe');
+      iframe.id = `letstalk-app-${app.id}`;
```

Comment:
this interpolation is used in multiple places, make it a method, and use a constant